### DISCUSSION TITLE
VS code recreates browse.vc.db file

### DISCUSSION THREAD

**Discussion by vgagrani** [microsoft/vscode-cpptools#11679] [General]
bvdev@429[~/src/browse/.vscode]#ls -lha
total 1.1G
drwxr-xr-x. 2 vivek.gagrani Engineering 4.0K Nov  7 17:33 ./
drwxr-xr-x. 8 vivek.gagrani Engineering 4.0K Nov  7 17:24 ../
-rw-r--r--. 1 vivek.gagrani Engineering 1.1G Nov  7 18:34 browse.vc.db
-rw-r--r--. 1 vivek.gagrani Engineering  556 Nov  7 01:07 c_cpp_properties.json
-rw-r--r--. 1 vivek.gagrani Engineering    0 Nov  7 18:34 .nfs0000000003f6090d00000040
-rw-r--r--. 1 vivek.gagrani Engineering  32K Nov  7 18:34 .nfs0000000003f6090e0000003f
-rw-r--r--. 1 vivek.gagrani Engineering  936 Nov  7 16:58 settings.json

bvdev@429[~/src/browse/.vscode]#ls -lha
total 1.1G
drwxr-xr-x. 2 vivek.gagrani Engineering 4.0K Nov 14 13:26 ./
drwxr-xr-x. 8 vivek.gagrani Engineering 4.0K Nov  7 17:24 ../
-rw-r--r--. 1 vivek.gagrani Engineering 4.8M Nov 14 13:29 browse.vc.db
-rw-r--r--. 1 vivek.gagrani Engineering  32K Nov 14 13:29 browse.vc.db-shm
-rw-r--r--. 1 vivek.gagrani Engineering 8.0M Nov 14 13:29 browse.vc.db-wal
-rw-r--r--. 1 vivek.gagrani Engineering  556 Nov  7 01:07 c_cpp_properties.json
-rw-r--r--. 1 vivek.gagrani Engineering 1.1G Nov 14 13:26 .nfs0000000003f6090800000051
-rw-r--r--. 1 vivek.gagrani Engineering 4.0M Nov 14 12:45 .nfs000000000a953eed00000050
-rw-r--r--. 1 vivek.gagrani Engineering  32K Nov 14 13:26 .nfs000000000a953eee0000004f
-rw-r--r--. 1 vivek.gagrani Engineering  938 Nov  7 19:20 settings.json

 everytime I remote login ssh to my system, intellisense recreate the vc.db from the scratch again. I have a big project alomost 11G , and intellisense starts creating the symbol database again whenever I login or if somehow my ssh goes to reconnect state. This is realluy time consuming. There are few instances already opened and closed for this and I followed whatever there on those incidents mentioned but still there is no luck. Some of the similar incidents reported are
[this](https://github.com/microsoft/vscode-cpptools/issues/9837) and [this](https://github.com/microsoft/vscode-cpptools/issues/3470) 
Below is my workspace settings.json file

     {
        "C_Cpp.default.browse.databaseFilename": "${workspaceFolder}/.vscode/browse.vc.db",
        "C_Cpp.errorSquiggles": "disabled",
        "cmake.configureOnOpen": false,
        "files.exclude": {
            "sigmatch/suricata-3.2.1/**": true,
            "sigmatch/suricata-4.1.0/**": true,
            "sigmatch/suricata-2.0/**": true,
            "sigmatch/suricata-3.0/**": true,
            "sigmatch/suricata-2.0.8/**": true,
            "sigmatch/suricata-2.0.3/**": true,
            "sigmatch/suricata-4.1.2/**": true,
            "sigmatch/suricata-4.1.3/**": true,
            "sigmatch/suricata-4.1.5/**": true,
            "sigmatch/suricata-5.0.4/**": true,
            "sigmatch/suricata-6.0.4/**": true,
            "sigmatch/suricata-6.0.5/**": true
        },
        //"C_Cpp.intelliSenseEngineFallback": "enabled",
       // "cmake.configureOnOpen": false,
       //"cmake.sourceDirectory": "/home2/vivek.gagrani/src/browse/6windgate/dpdk-pmd-mellanox-rdma-core
     }

bvdev@429[~/src/browse/.vscode]#cat c_cpp_properties.json

    {
        "configurations": [
            {
                "name": "Linux",
                "includePath": [
                    "${workspaceFolder}/**"
                ],
                "defines": [],
                "compilerPath": "/usr/bin/gcc",
                "cStandard": "c11",
                "cppStandard": "c++14",
                "browse": {
                    "limitSymbolsToIncludedHeaders": true,
                    "path": [
                        "${workspaceFolder}/**"
                     ]
                 },
                "intelliSenseMode": "linux-gcc-x64"
            }
        ],
        "version": 4
    }

 code --list-extensions --show-versions
        mjcrouch.perforce@4.15.7 
        ms-vscode.cmake-tools@1.15.31
        ms-vscode.cpptools@1.17.5
        ms-vscode.cpptools-extension-pack@1.3.0
        twxs.cmake@0.0.17

Can someone please help me on this

**Comment by sean-mcmanus** [comment]
@vgagrani With C_Cpp.loggingLevel set to "Debug" what does the logging show in regards to the database being reset? It's supposed to show an error message about why the database is being reset.

**Reply by JyChang012** [reply]
Hi, I have the same issue on a cluster with shared filesystem.
I have a large cpp project on an HPC cluster's shared filesystem and I have to start vscode remote on a new node each time I want to work on it. I find that the `browse.vc.db` file was regenerated each time I switch to a different node before intelliSense is available, which takes hours. How to prevent vscode cpptools from re-generating the `browse.vc.db` each time?
